walkthroughs/howto-mutual-tls-acm/README.md

```diff
@@ -0,0 +1,277 @@
+
+In this guide, we will be configuring Envoy proxies using certificates hosted in AWS Secrets Manager, which a modified Envoy image will retrieve during startup. We will have a virtual gateway connected to a single backend service. Both the gateway and backend proxies will present certificates signed by the same Certificate Authority (CA), though you could choose to use separate CAs.
+
+## Prerequisites
+
+1. Install Docker. It is needed to build the demo application images.
+2. Install the unix command line utility `jq`. If you don't already have it, [you can install it from here](https://stedolan.github.io/jq/).
+3. Install session manager plugin from [here](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html). If that is not possible, use the AWS console to run commands from bastion host.
+
+## Part 1: Setup
+
+### Step 1: Create Color App Repository
+
+We'll start by setting up the basic docker images for our services. All commands will be provided as if run from the same directory as this README.
+
+First, we need to set a few environment variables before provisioning the infrastructure. Please change the value for `AWS_ACCOUNT_ID` and `ENVOY_IMAGE` below.
+
+```bash
+export AWS_ACCOUNT_ID=<your account id>
+export AWS_DEFAULT_REGION=us-west-2
+export ENVIRONMENT_NAME=mtls-ec2-appmesh-example
+export MESH_NAME=mtls-ec2-appmesh
+export SERVICES_DOMAIN="mtls-ec2.svc.cluster.local"
+export ENVOY_IMAGE=<get the latest from https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html>
+export COLOR_TELLER_IMAGE_NAME="colorteller"
+export ENVOY_IMAGE_NAME="colorteller-envoy"
+```
+
+Now let's create the ECR repositories.
+
+```bash
+./infrastructure/ecr-repositories.sh
+```
+Next, build and deploy the color app image.
+
+```bash
+GO_PROXY=direct ./src/colorteller/deploy.sh
+```
+
+Note that the example apps use go modules. If you have trouble accessing https://proxy.golang.org during the deployment you can override the GOPROXY by setting `GO_PROXY=direct` as shown above.
+
+### Step 2: Export our Custom Envoy Image
+
+Finally, we can build and deploy our custom Envoy image. This container has a `/keys` directory and a custom startup script that will pull the necessary certificates from `AWS Secrets Manager` before starting up the Envoy proxy.
+
+```bash
+./src/customEnvoyImage/deploy.sh
+```
+
+> Note: This walkthrough uses this custom Envoy image for illustration purposes; using this method, rotating certificates will require a deployment of your tasks. You may instead choose to mount to your tasks an Elastic Block Storage or Elastic File System volume to store your TLS materials and reference them through the API accordingly.
```

Comment:
> ...using this method, rotating certificates will require a deployment of your tasks.

This is true regardless of the method you choose. App Mesh does not support an integration with ACM for mTLS at this time. The only way to consume ACM certificates is to export them and treat them like file provided certificates (as you've done in this walkthrough).

Please change to something like this:

> App Mesh does not support an integration with ACM for mTLS at this time. These instructions are for exporting an ACM certificate and using it as a customer file provided certificate. Your certificates will not be automatically updated after renewal, rotating certificates will require a restart or deployment of your tasks...

Lets also put a note at the top of this README.

Reply:
CloudWatch has an event trigger for notifying users on expiration of certs. That can be used to trigger an automation to update the service with `force-new-deployment` option.